kernel/univ.ml

```diff
@@ -38,22 +38,34 @@ struct
   open Names
 
   module UGlobal = struct
-    type t = DirPath.t * string * int
-
-    let make dp s i = (DirPath.hcons dp,s,i)
-
-    let repr x : t = x
-
-    let equal (d, s, i) (d', s', i') = Int.equal i i' && DirPath.equal d d' && String.equal s s'
-
-    let hash (d,s,i) = Hashset.Combine.combine3 i (String.hash s) (DirPath.hash d)
-
-    let compare (d, s, i) (d', s', i') =
-      if i < i' then -1
-      else if i' < i then 1
-      else let c = DirPath.compare d d' in
-        if not (Int.equal c 0) then c
-        else String.compare s s'
+    type t = {
+      library : DirPath.t;
+      process : string;
+      uid : int;
+    }
+
+    let make dp s i = {
+      library = dp;
+      process = s;
+      uid = i;
+    }
+
+    let repr x = (x.library, x.process, x.uid)
+
+    let equal u1 u2 =
+      Int.equal u1.uid u2.uid &&
+      DirPath.equal u1.library u2.library &&
+      String.equal u1.process u2.process
+
+    let hash u = Hashset.Combine.combine3 u.uid (String.hash u.process) (DirPath.hash u.library)
+
+    let compare u1 u2 =
```

Comment:
Maybe you should add a case to exiting `OrderedType.Pair` to also have `OrderedType.Triple` ?

Reply:
Tho I prefer base approach where you just have n-ary `compare` builders.

Reply:
Not sure I understand what you're asking for.

Reply:
Sorry I was not clear. I mean that we already have `OrderedType.Pair` which will allow to build a compare instance for a tuple, so I was wondering if we should extend that.

My second comment questions if instead of using a functor, we should just use helpers `compare2 : 'a cmp -> 'b cmp -> ('a * 'b) cmp` to build that boilerplate.

Reply:
For this kind of use I think it's better to use a named record rather than an anonymous tuple. So it doesn't make sense to rely on a generic function.

> we already have OrderedType.Pair

Do we? I was not aware of that. Where is it supposed to be found, I can't find it grepping the code.

Reply:
`clib/orderedType.mli`